### DISCUSSION TITLE
parseMessage (common.ts) throws and caches error on each socket message

### DISCUSSION THREAD

**Discussion by FreeRiderBysik** [enisdenjo/graphql-ws#395] [General]
I'm using  graphql-ws with uWebSockets and  @urql.

I discovered tons of errors like `Message is expected to be an object, but got string`, then launching debugger 
(Stack is:  onmessage->parseMessage->validateMessage).
So, my socket messages are strings, it working correct. Now parseMessage looks like:
```
try {
    return validateMessage(data);
  } catch {
    if (typeof data !== 'string') {
      throw new Error('Only strings are parsable messages');
    }
    const message = JSON.parse(data, reviver);
    return validateMessage(message);
  }
```

So it will throw Error and then catch it every time it got string. It is antipattern (exception as flow control) and it can effect performance. 
I suppose it will be simpler and work better with something like this:
```
const message = (typeof data === 'string') ? JSON.parse(data, reviver) : data;
return validateMessage(message);  
 ```

![image](https://user-images.githubusercontent.com/1512168/188662798-c21c64c2-76ca-400b-a0af-ba1254be9efb.png)

**Comment by btravel** [comment]
Hi,

Looks really strange, especially if consider that stringify is forced on the server side:

https://github.com/enisdenjo/graphql-ws/blob/master/src/server.ts#L691

https://github.com/enisdenjo/graphql-ws/blob/master/src/common.ts#L455

**Comment by enisdenjo** [comment]
@FreeRiderBysik you're absolutely right, the parsing was too convoluted. I pushed [a fix following you suggestion](https://github.com/enisdenjo/graphql-ws/commit/d44c6f18ccd2bc539bb541e6bfd1bf0c769c7856) and have released it with [v5.10.2](https://github.com/enisdenjo/graphql-ws/releases/tag/v5.10.2).

Thanks for investigating and making a suggestion! Would appreciate it if you could try again and report any other discrepancies.
